js/main.js

```diff
@@ -0,0 +1,144 @@
+
+const canvas = document.getElementById('wordSearchCanvas');
+const context = canvas.getContext('2d');
+
+
+function redraw() {
+  let height = canvas.getBoundingClientRect().height;
+  let width = canvas.getBoundingClientRect().width;
+
+  // console.log('Height is: ' + height + ' Width is: ' + width);
+}
+
+/**
+ * Returns a word seach game board that is created based of the dimensions
+ * of the allocated screen space with each letter being allocated it's
+ * own square within the board.
+ * @param {number} canvasHeight
+ * @param {number} canvasWidth
+ * @returns {Object}
+ */
+function createGameBoard(canvasHeight, canvasWidth) {
+  let gameBoard =
+      {startPointX: 0, startPointY: 0, gameHeight: 10, gameWidth: 10, board: []}
+
+  let heightScaler = canvasHeight / gameBoard.gameHeight;
+  let widthScaler = canvasWidth / gameBoard.gameWidth;
+
+  if (widthScaler > heightScaler) {
+    widthScaler = heightScaler;
+  } else {
+    heightScaler = widthScaler;
+  }
+
+  let middlePoint = canvasWidth / 2 - (widthScaler * gameBoard.gameWidth / 2);
+  gameBoard.startPointX = middlePoint;
+
+  currX = gameBoard.startPointX;
+  currY = gameBoard.startPointY;
+
+  for (let i = 0; i < gameBoard.gameHeight; ++i) {
+    let rowOfSquares = [];
+    for (let j = 0; j < gameBoard.gameWidth; ++j) {
+      let square = {
+        xCoord: currX,
+        yCoord: currY,
+        width: widthScaler,
+        height: heightScaler,
+        character: '-'
+      };
+
+      rowOfSquares.push(square);
+
+      currX += widthScaler;
+    }
+    gameBoard.board.push(rowOfSquares);
+    currX = gameBoard.startPointX;
+    currY += heightScaler;
+  }
+
+  return gameBoard;
+}
+
+/**
+ * Converts ranNumber from a float to a character in the
+ * english alphabet based on what interval it is within the charFrequency.
+ * @param {array} charFrequency an array of floats where element at index i
+ *    less element index i - 1 is the frequency that the i'th letter occurs in
+ *    the english language
+ * @param {number} ranNumber a random float between 0-101.432
+ * @return {String}
+ */
+function getCharater(charFrequency, ranNumber) {
+  let index = 0;
+
+  for (let i = 0; i < 26; ++i) {
+    if (ranNumber < charFrequency[i]) {
+      index = i;
+      break;
+    }
+  }
+
+  return String.fromCharCode(97 + index)
+}
+
+/**
+ * Based on a weighted random distribution, it populates the wordSearchBoard
+ * with random letters.
+ * @param {object} wordSearchBoard
+ */
+function randomizeCharacters(wordSearchBoard) {
+  /** @const {number} */
+  const frequency = [
```

Comment:
Given that this is a constant - move it to global scope

```js
/** @const {!Array<number>} */
const LETTER_FREQUENCIES = [ ... ];
```

Reply:
Updated